packages/dds/tree/src/feature-libraries/chunked-forest/chunkedForest.ts

```diff
@@ -0,0 +1,360 @@
+            path = path.parent;
+        }
+        cursorToMove.setToAboveDetachedSequences();
+        while (keyStack.length > 0) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            cursorToMove.enterField(keyStack.pop()!);
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            cursorToMove.enterNode(indexStack.pop()!);
+        }
+    }
+}
+
+class Cursor extends BasicChunkCursor implements ITreeSubscriptionCursor {
+    public constructor(
+        public readonly forest: ChunkedForest,
+        public state: ITreeSubscriptionCursorState,
+        siblingStack: SiblingsOrKey[],
+        indexStack: number[],
+        indexOfChunkStack: number[],
+        indexWithinChunkStack: number[],
+        siblings: SiblingsOrKey,
+        index: number,
+        indexOfChunk: number,
+        indexWithinChunk: number,
+    ) {
+        super(
+            siblingStack,
+            indexStack,
+            indexOfChunkStack,
+            indexWithinChunkStack,
+            siblings,
+            index,
+            indexOfChunk,
+            indexWithinChunk,
+        );
+    }
+
+    public setToAboveDetachedSequences(): void {
+        this.siblingStack.length = 0;
+        this.indexStack.length = 0;
+        this.indexOfChunkStack.length = 0;
+        this.indexWithinChunkStack.length = 0;
+        this.siblings = [this.forest.roots];
+        this.index = 0;
+        this.indexOfChunk = 0;
+        this.indexWithinChunk = 0;
+    }
+
+    public fork(): Cursor {
+        // Siblings arrays are not modified during navigation and do not need be be copied.
+        // This allows this copy to be shallow, and `this.siblings` below to not be copied as all.
+        return new Cursor(
+            this.forest,
+            this.state,
+            [...this.siblingStack],
+            [...this.indexStack],
+            [...this.indexOfChunkStack],
+            [...this.indexWithinChunkStack],
+            this.siblings,
+            this.index,
+            this.indexOfChunk,
+            this.indexWithinChunk,
+        );
+    }
+
+    public buildFieldAnchor(): FieldAnchor {
+        const path = this.getFieldPath();
+        const anchor =
+            path.parent === undefined ? undefined : this.forest.anchors.track(path.parent);
+        return { parent: anchor, fieldKey: path.field };
+    }
+
+    public free(): void {
+        this.state = ITreeSubscriptionCursorState.Freed;
+    }
+
+    public buildAnchor(): Anchor {
+        return this.forest.anchors.track(this.getPath());
+    }
+
+    public getPath(): UpPath {
+        return super.getPath() ?? fail("no path when at root");
+    }
+
+    public clear(): void {
+        this.state = ITreeSubscriptionCursorState.Cleared;
+        this.setToAboveDetachedSequences();
+    }
+}
+
+/**
+ * @returns an implementation of {@link IEditableForest} with no data or schema.
```

Comment:
nit: This is something I want to discuss a bit more but after seeing what the generated API docs actually look like, I no longer think that tags make sufficient API docs. Mainly because tags don't show up in consolidated API pages like the main package page that lists what's in the package. 

As an example, here's what the `buildForest` description looks like on the main page for the tree package:

![image](https://user-images.githubusercontent.com/16496050/213784000-811c457c-87fa-4868-b243-f89982262abe.png)

I think it'd be more helpful if we add some context on when it should be used or something, especially since it's a public facing API. 

I also suspect that the naming of the functions will conflict at some point if we're planning on allowing customers to use both while not exposing the underlying forests.

Reply:
I picked this name assuming we wound package export one forestBuilder. Now that I think about it we might not need to package export any forest builders as they all come from the one SharedTree constructs or forks of it, so I have done the rename.

As for the documentation issue, personally I'd prefer to fix API Extractor (and what ever other tools need fixing), since the current behavior isn't good (the API reports list members with inherited documentation or docs that are just tags as "(undocumented)" for example, see https://github.com/microsoft/rushstack/issues/3462).

Reply:
The inherited thing is definitely weird, but I am thinking that the case where it's just tags may be justified since there are pages where it certainly does look like it's undocumented.

I think we can definitely fix api extractor though, we are just using a fork of it anyways. Anyways, this is probably a conversation for a different setting so this issue can just be resolved.